### ISSUE TITLE
amp-selector: Propagate trust from toggle/selectUp/selectDown actions to "select" event

### ISSUE THREAD

**Issue by sebastianbenz** [ampproject/amphtml#24425]
## Proposed deprecation

When an `amp-selector`'s `toggle`, `selectUp`, or `selectDown` action is invoked, change the  trust level of the resulting `select` event. 

Specifically, change it from always having "high trust" to having the same trust as the originating event that triggered `toggle` etc.

For example:
```html
<amp-selector id=foo on="select: ..."></amp-selector>

<button on="tap:foo.selectUp">
  Propagate "high trust" to amp-selector.select via `tap`
</button>

<!-- Propagates "low trust" to amp-selector.select via `timeUpdate`. -->
<amp-video on="timeUpdate:foo.selectUp" src=...></amp-video>
```

## What's the issue?

It's possible to circumvent AMP's low-trust level restrictions by combining amp-selector's low trust `toggle` action with its high-trust `select` event.

### Executing amp-bind on page load

For example. it's currently possible to automatically evaluate amp-bind expressions on page load by combining `amp-carousel` and `amp-selector`. The basic approach is: 

1. Change amp-selector selection on a `slideChange` event: `on="slideChange:ampSelector.toggle(index=1, value=true)"` (directly calling `AMP.setState({})` is not allowed).
2. Listen to the `select` event in the amp-selector and then call `AMP.setState()` (which is allowed).

See an example here:

https://amp-playground.appspot.com/amp/5644009611460608

This example exposes two problems:

1. `amp-carousel` v0.2 fires the `slideChange` event on page load (not sure if this is intended). Even it is not, adding the `autoplay` flag would expose the same behavior.
2. A feature interaction between `amp-carousel` and `amp-selector`, which allows circumventing the `AMP.setState()` restriction.

### Executing amp-bind on scroll

Here is another example demonstrating how to set AMP state via amp-position-observer. It uses the same approach as before, but this time amp-position-observer triggers a state change via amp-selector on scroll:

```
<amp-position-observer
    on="enter:ampSelector.toggle(index=0, value=true)"
    layout="nodisplay">
  </amp-position-observer>
<amp-selector id="ampSelector" on="select:AMP.setState({changed: true})">
    <div option></div>
</amp-selector>
<div hidden [hidden]=!changed>I should not be visible</div>
```

https://amp-playground.appspot.com/amp/5727517264576512 

## What browsers are affected?

All browsers

## Which AMP version is affected?

AMP ⚡ HTML – Version 1909051756330 and older

**Comment by choumx** [comment]
Nice catch.

**Comment by sparhami** [comment]
amp-carousel 0.2 fires a `slideChange` on page load, but it should be a low trust event. This is intended behavior. It also fires one on autoplay, which is again low trust.

This contrasts with amp-carousel 0.1, which fires a `slideChange` on autoplay, as a high trust action.

We intended to change this for amp-carousel 0.1 (#21171), but have decided to postpone the change until `amp-inline-gallery` is released. One thing to note is that we gave a six week deprecation notice for the change. While it was not intentional to fire a high-trust `slideChange` from autoplay, developers did depend on it, which merited a deprecation notice.

I believe the same is true here. We should first give a deprecation notice and give developers ample time to change their design and adapt rather than immediately breaking their pages.

**Comment by sparhami** [comment]
Also, it doesn't seem correct that amp-selector should be able to toggle via a low trust action when it doesn't have a size defined layout (`layout="container"`). This means any low trust action (such as page scrolling via position observer) can cause the page layout to jump, which should not be the case.

Contrast this with carousel, which allows you to change slides on low trust action, but requires a defined size, so this cannot cause the page to jump.

As far as I know, amp-selector has the only low trust action in AMP that can affect page layout. My understanding of action trust is that this should not be the case.

**Comment by choumx** [comment]
I wonder if amp-selector is ever used with non-container layout in practice.

**Comment by nainar** [comment]
Agree with the deprecation plan. 
Also filed an issue to make sure we cover what high/low trust means on amp.dev here: https://github.com/ampproject/amphtml/issues/24441
